homeassistant/components/amberelectric/manifest.json

```diff
@@ -6,5 +6,5 @@
   "documentation": "https://www.home-assistant.io/integrations/amberelectric",
   "iot_class": "cloud_polling",
   "loggers": ["amberelectric"],
-  "requirements": ["amberelectric==1.1.1"]
+  "requirements": ["amberelectric==2.0.2"]
```

Comment:
Why does the library both have a pyproject.toml and a setup.py? The requirement specifiers are not exactly the same in both files. I'd recommend removing the `setup.py` file and just use a pyproject.toml file.

Reply:
I've synced them up. The generator always has one, and I can't figure out how to remove it, but at least they are the same now.

Reply:
Can you remove the build settings from pyproject.toml then so it's clear what settings will be used to build the project?

Reply:
Removing setup.py breaks the build process. And I'm not entirely sure how this impacts the home assistant integration?

Reply:
There are still dependencies specified in both pyproject.toml and setup.py. This is the problem that I'd suggest to address. When reading the project source it's not clear what dependencies the project uses. Only one set of dependencies in either file should be needed for the production build, right?

Reply:
Have worked out the correct build process using `poetry` and have removed the setup.py file.

Reply:
Version 2.0.5 published on PyPI doesn't correspond to the version published on GitHub. The one on PyPI still has a setup.py file.

Please publish a new version, eg 2.0.6, on both GitHub and PyPI that comes from the exact same source.

I'd also recommend adding a GitHub action to automatically publish a release on PyPI after a release is tagged on GitHub. That way you don't need to worry about doing that manually. Nowadays it's even possible to verify the supply chain via digital attestations when using the `pypa/gh-action-pypi-publish` GitHub action.

https://github.com/pypa/gh-action-pypi-publish